src/components/Room/UsernameForm.tsx

```diff
@@ -11,31 +11,29 @@ import { useDispatch, useSelector } from 'react-redux';
 import { setUsername } from '../../store/playerSlice';
 import { setHost } from '../../store/gameSlice';
 import './userForm.css';
-
+import { useFormik } from 'formik';
+import * as yup from 'yup';
 interface UsernameFormProps {
-  handleClose: Function;
+  handleClose: (bool: boolean) => void;
 }
-
+const validationSchema = yup.object({
+  username: yup.string().required('Username is required'),
+});
 function UsernameForm({ handleClose }: UsernameFormProps) {
   const dispatch = useDispatch();
   const { roomId } = useParams() as { roomId: string };
 
   const { playerId, username, isHost } = useSelector((state: RootState) => state.player);
 
   // If we have a username in redux slice, start it with that. else, start it with empty string.
-  const [inputtedUsername, setInputtedUsername] = useState<string>(username ? username : '');
   const [usernameSubmissionDone, setUsernameSubmissionDone] = useState<boolean>(false);
-
-  const submitHandler = async (event: React.MouseEvent<HTMLButtonElement>) => {
-    event.preventDefault();
-
-    // TODO: Validation - may want to use formik/yup?
-    // Make sure the username they gave isn't empty / made of only white spaces
-    const trimmedInputtedUsername: string = inputtedUsername.trim();
-    if (trimmedInputtedUsername === '') {
-      console.log('cannot have empty username');
-      // stop now! and display error. user needs to resubmit. - Or use formik/yup
-    } else {
+  const formik = useFormik({
+    initialValues: {
+      username: username || '',
```

Comment:
This alone did not fix this for me - it's because when this useFormik runs for the first time we haven't gotten the username from redux, so it never reruns - i googled around and found something that lets one pull in state to start a formik form with.

```js
  const formik = useFormik({
    enableReinitialize: true,
    initialValues: {
      username: username ? username : '',
    },
    // etc....
```
should do the trick